plugins/packages/couchdb/lib/index.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  QueryError,
+  QueryResult,
+  QueryService,
+  ConnectionTestResult,
+} from "@tooljet-plugins/common";
+import { SourceOptions, QueryOptions } from "./types";
+import got from "got";
+const JSON5 = require('json5')
+export default class Couchdb implements QueryService {
+  async run(
+    sourceOptions: SourceOptions,
+    queryOptions: QueryOptions
+  ): Promise<QueryResult> 
+  {
+    
+    let result = {};
+    let response = null;
+    const {
+      operation,
+      record_id,
+      limit,
+      view_url,
+      start_key,
+      end_key,
+      skip,
+      descending,
+    } = queryOptions;
+    const { username, password, port, host, database, protocol } =
+      sourceOptions;
+    const revision_id = queryOptions.rev_id;
+
+    const authHeader = () => {
+      const combined = `${username}:${password}`;
+      const key = Buffer.from(combined).toString("base64");
+      return { Authorization: `Basic ${key}` };
+    };
+
+    try {
+      switch (operation) {
+        case "list_records": {
+          response = await got(
+            `${protocol}://${host}:${port}/${database}/_all_docs`,
+            {
+              method: "get",
+              headers: authHeader(),
+              searchParams: {
+                ...(limit?.length > 0 && { limit }),
+                ...(skip?.length > 0 && { skip }),
+                ...(descending && { descending }),
+              },
+            }
+          );
+          result = JSON5.parse(response.body);
+          break;
+        }
+
+        case "retrieve_record": {
+          response = await got(
+            `${protocol}://${host}:${port}/${database}/${record_id}`,
+            {
+              headers: authHeader(),
+              method: "get",
+            }
+          );
+          result = JSON5.parse(response.body);
+          break;
+        }
+
+        case "create_record": {
+          response = await got(`${protocol}://${host}:${port}/${database}`, {
+            method: "post",
+            headers: authHeader(),
+            json: {
+              records: JSON5.parse(queryOptions.body),
+            },
+          });
+          result = JSON5.parse(response.body);
```

Comment:
No need to parse response using `json5` parse user inputs like `queryOptions.body`. Create a private function to parse json
refer https://github.com/ToolJet/ToolJet/blob/develop/plugins/packages/bigquery/lib/index.ts